crates/matrix-sdk/src/sliding_sync/builder.rs

```diff
@@ -63,13 +65,32 @@ impl SlidingSyncBuilder {
 
     /// Add the given list to the lists.
     ///
-    /// Replace any list with the name.
+    /// Replace any list with the same name.
     pub fn add_list(mut self, list: SlidingSyncList) -> Self {
         self.lists.insert(list.name().to_owned(), list);
-
         self
     }
 
+    /// Enroll the list in caching, reloads it from the cache if possible, and
+    /// adds it to the list of lists.
+    ///
+    /// This will raise an error if a [`storage_key()`] was not set, or if there
+    /// was a I/O error reading from the cache.
+    ///
+    /// Replace any list with the same name.
+    pub async fn add_cached_list(mut self, mut list: SlidingSyncListBuilder) -> Result<Self> {
+        let Some(ref storage_key) = self.storage_key else {
+            return Err(super::error::Error::MissingStorageKeyForCaching.into());
+        };
+        let reloaded_rooms = list.set_cached_and_reload(&self.client, storage_key).await?;
+        for (key, frozen) in reloaded_rooms {
+            self.rooms
+                .entry(key)
+                .or_insert_with(|| SlidingSyncRoom::from_frozen(frozen, self.client.clone()));
+        }
+        Ok(self.add_list(list.build()))
```

Comment:
Typical case where https://github.com/matrix-org/matrix-rust-sdk/pull/1758 changes the behavior of `add_list` which now takes a list builder.